python/tests/io/test_base.py

```diff
@@ -0,0 +1,188 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import tempfile
+
+import pytest
+
+from iceberg.io.base import FileIO, InputFile, OutputFile
+
+
+class LocalInputFile(InputFile):
+    """An InputFile implementation for local files (for test use only)"""
+
+    def __init__(self, location: str):
+        if not location.startswith("file://"):
+            raise ValueError("LocalInputFile location must start with `file://`")
+        super().__init__(location=location.split("file://")[1])
+
+    def __len__(self):
+        return os.path.getsize(self.location)
+
+    def exists(self):
+        return os.path.exists(self.location)
+
+    def open(self):
+        return open(self.location, "rb")
+
+
+class LocalOutputFile(OutputFile):
+    """An OutputFile implementation for local files (for test use only)"""
+
+    def __init__(self, location: str):
+        if not location.startswith("file://"):
+            raise ValueError("LocalOutputFile location must start with `file://`")
+        super().__init__(location=location.split("file://")[1])
+
+    def __len__(self):
+        return len(self._file_obj)
```

Comment:
Should this be `os.path.getsize(self.location)`? I don't see any other reference to `_file_obj`.

Reply:
That's right thanks! I updated it to use the parsed uri added in another commit so it's `os.path.getsize(self.parsed_location.path)` now. I also added validation of `len` in the tests.

relevant commit: [7b625cf](https://github.com/apache/iceberg/pull/3691/commits/7b625cfee1937ab2b9dd9327ee09b940014d2ba3)